### PR TITLE
Improved UNet 46408211

### PR DESCRIPTION
**Pull request by HyperCactus**
# Segmentation of skin lesions using an Improved UNet
##### Author: Sam Pollard, 46408211
I have done task 1, segmentation of the ISIC 2017 dataset using the Improved UNet.
My repository contains the following files:
* `dataset.py` contains the custom PyTorch data set for the ISIC 2017 data.
* `modules.py` contains the Improved UNet model class as well as other modules used in the network architecture.
* `dice_loss.py` Contains a custom loss function module based on the dice similarity coefficient.
* `global_params.py` contains global variables such as file directories for the data and hyperparameters for training.
* `utils.py` contains helper functions for visualizing results, saving and loading models, etc.
* `train.py` the training script for the model.
* `predict.py` loads the trained model and visualizes example usage as well as calculating evaluation metrics.
* See `requirements.yaml` for dependencies.

I was able to attain a dice similarity coefficient of approximately 0.83 after a combined 70 epochs of training.

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
This is an initial inspection, no action is required at this point.

- Well organized files, but wrong files path. All student constructed document should be packed in one document under the ‘recognition’ document.
- Algorithm solves the problem appropriately. Average Dice Coefficient in testing dataset: 0.8258. Predicted Mask output looks good. No data augmentation. 
- Correctly using PyTorch to construct the Improved U-Net models and functions.
- Good coding design. 
- Sufficient docstring and comment in the code, good references. 
- Easy difficulty problem, no higher difficulty marks.

**Comment by HyperCactus** [comment]
Update: Changed file structure as per feedback. Moved all project files and directories into recognition/UNet46408211

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage, could have used augmentation -1
Difficulty: Easy -10
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable, no usage info -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good

**Comment by shakes76** [comment]
Merging requires that repo README be restored. Does not affect grade but only the merge, can you fix?

**Comment by HyperCactus** [comment]
Restored repo README

**Comment by wangzhaomxy** [comment]
Feedback attempt and no feedback marks lost.
